CONTRIBUTING.md

```diff
@@ -141,14 +141,22 @@ Avoid preprocessor macros as much as possible (use enum not #define).
 Preprocessor usage should be reserved for expressing configure time
 options.
 
+Use std::shared_ptr and std::unique_ptr to avoid use of raw pointers
+in any interfaces or class members.  There should be no calls to new
+or delete, and instead to std::make_shared() and geopm::make_unique().
+Prefer to forward declare classes and structures in header files and
+avoid including local header files in other header files.
+
 Introducing a new file requires a license comment in its header with a
-corresponding copying_headers/header.* file.  The new file path must
-be listed in the corresponding copying_headers/MANIFEST.* file.  This
-can be tested by running the copying_headers/test_license script after
-committing the new file to git, removing the geopm/MANIFEST file and
-running the autogen.sh script.  Files for which a license comment is
-not appropriate should be listed in copying_headers/MANIFEST.EXEMPT.
-Any new installed files should also be added geopm.spec.in.
+corresponding copying_headers/header.* file.  If the new file does not
+have the legal text from copying_headers/header.BSD3-intel at the top
+of the file, the new file path must be listed in the corresponding
+copying_headers/MANIFEST.* file.  This can be tested by running the
+copying_headers/test_license script after committing the new file to
+git, removing the geopm/MANIFEST file and running the autogen.sh
+script.  Files for which a license comment is not appropriate should
+be listed in copying_headers/MANIFEST.EXEMPT.  Any new installed files
+should also be added geopm.spec.in.
```

Comment:
Depending on what is being installed, might need different changes in the Makefile.am.  Perhaps a list similar to the one below for man pages?